compiler/quilt/nodes.py

```diff
@@ -139,6 +140,38 @@ def _data(self):
 
         return store.load_dataframe(hash_list)
 
+def _create_filter_func(filter_dict):
+    filter_name = filter_dict.pop('name', None)
+    if filter_name is not None and not isinstance(filter_name, string_types):
+        raise ValueError("Invalid 'name'")
```

Comment:
probably worth printing the invalid thing too

Reply:
And do we want to say `Invalid 'name' in filtering dict`. I am frankly unclear as to why a name key would always be defined btw.

Reply:
Ok - but not sure what you mean by "why a name key would always be defined"?